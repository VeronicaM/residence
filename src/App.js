import React, { useState } from "react";

import "./App.scss";

function App() {
  const defaultState = {
    residences: [],
  };

  const [state, setState] = useState(defaultState);

  return (
    <div className="app__container">
      <header>
        <h1> Residence App </h1>
      </header>
      <main>
      </main>
      <footer>Created by @Veronica Mihai 2020 with â˜• and ðŸŽµ</footer>
    </div>
  );
}

export default App;
