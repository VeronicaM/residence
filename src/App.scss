@import "./core/styles/";

.app__container {
  display: flex;
  flex-direction: column;
  padding: $page-spacing;

  @media screen and (max-width: 768px) {
    padding: $page-spacing-mobile;
  }
}

.app__title {
  text-align: center;
}

.main__container {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
}

.loader__container {
  align-items: center;
  display: flex;
  height: 90vh;
  justify-content: center;
}
