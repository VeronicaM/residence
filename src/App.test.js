import { render, screen } from '@testing-library/react';
import App from './App';

xtest('renders header text', () => {
  render(<App />);
  const headerText = screen.getByText(/Residence App/i);
  expect(headerText).toBeInTheDocument();
});
