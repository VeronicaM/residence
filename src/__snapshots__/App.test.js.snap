// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the component is rendered correctly with the passed in data 1`] = `
<div>
  <div
    class="app__container"
  >
    <header>
      <h1
        class="app__title"
      >
         Residence App 
      </h1>
    </header>
    <main
      class="main__container"
    >
      <form
        class="c-row residence-form__container"
      >
        <div
          class="c-field"
        >
          <label
            class="c-label"
            for="zipCode"
          >
            <strong>
               
              ZIP Code
              :
            </strong>
            <input
              class="c-input"
              data-testid="custom-input"
              name="zipCode"
              type="text"
              value=""
            />
          </label>
        </div>
        <div
          class="c-field"
        >
          <label
            class="c-label"
            for="number"
          >
            <strong>
               
              Residence Number
              :
            </strong>
            <input
              class="c-input"
              data-testid="custom-input"
              name="number"
              type="number"
              value=""
            />
          </label>
        </div>
        <div
          class="c-field"
        >
          <label
            class="c-label"
            for="latitude"
          >
            <strong>
               
              Latitude
              :
            </strong>
            <input
              class="c-input"
              data-testid="custom-input"
              name="latitude"
              type="number"
              value=""
            />
          </label>
        </div>
        <div
          class="c-field"
        >
          <label
            class="c-label"
            for="longitude"
          >
            <strong>
               
              Longitude
              :
            </strong>
            <input
              class="c-input"
              data-testid="custom-input"
              name="longitude"
              type="number"
              value=""
            />
          </label>
        </div>
        <div
          class="c-field"
        >
          <label
            class="c-label"
            for="noResidents"
          >
            <strong>
               
              Number of Residents
              :
            </strong>
            <input
              class="c-input"
              data-testid="custom-input"
              name="noResidents"
              type="number"
              value=""
            />
          </label>
        </div>
        <button
          class="c-cta"
          disabled=""
          type="submit"
        >
          Add
        </button>
      </form>
      <div>
        Loading...
      </div>
    </main>
    <footer>
      Created by @Veronica Mihai 2021 with â˜• and ðŸŽµ
    </footer>
  </div>
</div>
`;
